modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java

```diff
@@ -2686,14 +2686,17 @@ at the moment path passed as string (basePath) from upper components can be both
 			ComposedSchema composedSchema = new ComposedSchema();
 
 			if (allOfArray != null) {
-
-				for (JsonNode n : allOfArray) {
-					if (n.isObject()) {
-						schema = getSchema((ObjectNode) n, location, result);
-						composedSchema.addAllOfItem(schema);
+				if (allOfArray.size() > 1) {
```

Comment:
if this is going to be a new behavior, you could add it as an option, as the option resolveCombinators